tomviz/vtkChartHistogram.h

```diff
@@ -0,0 +1,70 @@
+/******************************************************************************
+
+  This source file is part of the tomviz project.
+
+  Copyright Kitware, Inc.
+
+  This source code is released under the New BSD License, (the "License").
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+******************************************************************************/
+#ifndef tomvizvtkChartHistogram_h
+#define tomvizvtkChartHistogram_h
+
+#include "vtkChartXY.h"
+
+#include "vtkNew.h"
+#include "vtkTransform2D.h"
+
+class vtkContextMouseEvent;
+class vtkCustomPiecewiseControlPointsItem;
+class vtkHistogramMarker;
+class vtkPiecewiseFunction;
+class vtkPiecewiseFunctionItem;
+class vtkPlotBar;
+class vtkScalarsToColors;
+class vtkTable;
+
+//-----------------------------------------------------------------------------
+class vtkChartHistogram : public vtkChartXY
+{
+public:
+  static vtkChartHistogram * New();
+
+  bool MouseDoubleClickEvent(const vtkContextMouseEvent &mouse) override;
+
+  // Set input for histogram
+  virtual void SetHistogramInputData(vtkTable* table, const char* xAxisColumn, const char* yAxisColumn);
+
+  // Set scalar visibility in the histogram plot bar
+  virtual void SetScalarVisibility(bool visible);
```

Comment:
Then it is fine, although I doubt this is going to be inherited from, so I wasn't really thinking about API that would be derived from. Perhaps premature, but at least clear.